source/common/grpc/common.cc

```diff
@@ -157,9 +220,12 @@ Http::MessagePtr Common::prepareHeaders(const std::string& upstream_cluster,
   message->headers().insertPath().value().append("/", 1);
   message->headers().insertPath().value().append(method_name.c_str(), method_name.size());
   message->headers().insertHost().value(upstream_cluster);
+  message->headers().insertTE().value().setReference(Http::Headers::get().TEValues.Trailers);
+  if (timeout) {
+    toGrpcTimeout(timeout.value(), message->headers().insertGrpcTimeout().value());
+  }
   message->headers().insertContentType().value().setReference(
       Http::Headers::get().ContentTypeValues.Grpc);
-  message->headers().insertTE().value().setReference(Http::Headers::get().TEValues.Trailers);
```

Comment:
Lost the comment thread here, sorry. Why did this move? If it needs to move (?) please add a comment so it doesn't get moved back.

Reply:
According to the gRPC/HTTP2 spec TE should be encoded before Timeout and ContentType,  see https://github.com/grpc/grpc/blob/master/doc/PROTOCOL-HTTP2.md. Will add a comment.